.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,115 @@
+name: Build and Test
+on: push
+      
+jobs:  
+  setup_build_ext:
+    name: Run setup build_ext
+    runs-on: ubuntu-latest
+    container: benjaminrueth/precice
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+      - name: Install & upgrade pip3
+        run: |
+          apt-get install -y python3-pip
+          pip3 install --upgrade --user pip
```

Comment:
Is it considered safe to mix the `pip` version distributed via the Ubuntu repository with a user-installed and updated version? I remember having troubles with that some time in the past (or a similar setup).

Reply:
As far as I know everything is fine as long as the `--user` flag is used. One thing that caused me a lot of trouble was `sudo pip3 ...`, since this changes the system pip.